cabal-install/Distribution/Client/IndexUtils.hs

```diff
@@ -179,6 +180,9 @@ readRepoIndex verbosity repo mode =
           RepoRemote{..} -> warn verbosity $
                "The package list for '" ++ remoteRepoName repoRemote
             ++ "' does not exist. Run 'cabal update' to download it."
+          RepoSecure{..} -> warn verbosity $
```

Comment:
May be worth refactoring the error message into a `errorPackageListDoesNotExist` function.